### PR TITLE
[#3868] - filter by published instead of looking at the doi, full_tex…

### PR DESCRIPTION
**Pull request by sblack-usu**
…t_search only once

### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [x] Peer Code review and approval

### Positive Test Case
1. navigate to /haspi and execute get /hsapi/resource/ with no extra parameters.  Verify the results return within a few seconds and are paginated.

### CODE REVIEW

File: hs_core/hydroshare/users.py [hunk 1]
```diff
@@ -338,7 +336,7 @@ def get_resource_list(creator=None, group=None, user=None, owner=None, from_date
         q.append(query)
 
     if published:
-        q.append(Q(doi__isnull=False))
+        q.append(Q(raccess__published=True))
```
Comment (sblack-usu) [inline hs_core/hydroshare/users.py:339]: Here is the piece to fix filtering by published.   The rest of the changes are just optimizations and cleanup.

File: hs_core/views/resource_rest_api.py [hunk 2]
```diff
@@ -299,7 +299,7 @@ def get_queryset(self):
         filter_parms['public'] = not self.request.user.is_authenticated()
         filtered_res_list = []
 
-        for r in hydroshare.get_resource_list(**filter_parms):
+        for r in self.paginate_queryset(hydroshare.get_resource_list(**filter_parms)):
```
Comment (sblack-usu) [inline hs_core/views/resource_rest_api.py:302]: This is the big win in optimization.  Before this, all resources returned in the queryset were being serialized, and then pagination was applied.  This change pages the results before serialization.

File: hs_core/hydroshare/users.py [hunk 3]
```diff
@@ -425,40 +423,21 @@ def get_resource_list(creator=None, group=None, user=None, owner=None, from_date
...
+        title_ids = Title.objects.filter(value__icontains=full_text_search).values_list('object_id', flat=True)
+
+        # Full text search must match within the title or abstract
+        if desc_ids:
+            flt = flt.filter(object_id__in=desc_ids)
+        elif title_ids:
+            flt = flt.filter(object_id__in=title_ids)
+        else:
+            # No matches on title or abstract, so treat as no results of search
+            flt = flt.none()
```
Comment (hyi) [inline hs_core/hydroshare/users.py:437]: Not sure why we need `flt = flt.none()``` here. Wouldn't it suffice to leave ```flt``` as is, i.e., remove the ```else:``` condition?

Comment (sblack-usu) [reply hs_core/hydroshare/users.py:437]: I simply moved this logic outside of the loop over the q array.  Looking this over, the description and title ids should be combined and then just applied to the filter.  I'll make the change.

Comment (sblack-usu) [reply hs_core/hydroshare/users.py]: @hyi - see the latest commit

**Pull request merged**